input_definition.go

```diff
@@ -254,6 +265,56 @@ type InputDefinitionInfo struct {
 	Fields []InputDefinitionField `json:"fields"`
 }
 
+// Validate the InputDefinitionInfo data
+func (i *InputDefinitionInfo) Validate(columnLabel string) error {
+	numPrimaryKey := 0
+	accountRowID := make(map[string]uint64)
+
+	if len(i.Frames) == 0 {
+		return fmt.Errorf("At least one frame is required per Input Definition")
+	}
```

Comment:
as discussed with Travis before, "frames and fields are both required (i suppose fields cold be empty, but in that case it would only be used to auto-create frames on startup). maybe for now we make them both required, and if we want to relax that later we can". I did that in createInputDefinition func (line 660-662 in index.go), but since we check it here, I think those lines can be remove and use ErrInputDefinitionAttrsRequired here insead of `fmt.Errorf("At least one frame is required per Input Definition")`.